js/render/draw_symbol.js

```diff
@@ -125,7 +125,8 @@ function drawSymbol(painter, layer, posMatrix, tile, elementGroups, prefix, sdf)
         texsize = [painter.glyphAtlas.width / 4, painter.glyphAtlas.height / 4];
     } else {
         painter.spriteAtlas.bind(gl, alignedWithMap || painter.options.rotating ||
-            painter.options.zooming || fontScale !== 1 || sdf || painter.transform.pitch);
+            painter.options.zooming || fontScale !== 1 || sdf || painter.transform.pitch ||
+            browser.devicePixelRatio !== painter.spriteAtlas.pixelRatio);
```

Comment:
No, I don't think that's any simpler. I was thinking that there might be some groupings of multiple clauses that would make sense semantically, but if that's not the case :ship:.